packages/grid_client/scripts/mass_deployments.ts

```diff
@@ -3,107 +3,201 @@ import {
   FarmFilterOptions,
   FilterOptions,
   generateString,
+  GridClient,
   MachineModel,
   MachinesModel,
   NetworkModel,
   randomChoice,
+  TwinDeployment,
 } from "../src";
 import { config, getClient } from "./client_loader";
 import { log } from "./utils";
 
+async function pingNodes(grid3: GridClient, nodes: any[]): Promise<any[]> {
+  const pingPromises = nodes.map(async node => {
+    const pingPromise = grid3.zos.pingNode({ nodeId: node.nodeId });
+    const timeoutPromise = new Promise((_, reject) =>
+      setTimeout(() => reject(new Error(`Timeout: Ping to node ${node.nodeId} took too long`)), 10000),
+    );
+
+    try {
+      const res = await Promise.race([pingPromise, timeoutPromise]);
+      return { node, res };
+    } catch (error) {
+      return { node, error };
+    }
+  });
+
+  const results = await Promise.all(pingPromises);
+
+  return results;
+}
+
 async function main() {
   const grid3 = await getClient();
+  grid3.clientOptions.deploymentTimeoutMinutes = 2;
 
   const errors: any = [];
+  const offlineNodes: number[] = [];
   let failedCount = 0;
   let successCount = 0;
+  const batchSize = 50;
+  const totalVMs = 250;
+  const batches = totalVMs / batchSize;
+
+  // resources
+  const cru = 1;
+  const mru = 256;
+  const diskSize = 1;
+  const rootFs = 1;
+  const publicIp = false;
+
+  console.time("Farms Time");
+  const farms = await grid3.capacity.filterFarms({
+    nodeMRU: mru / 1024,
+    nodeSRU: diskSize + rootFs,
+    publicIp: publicIp,
+    availableFor: await grid3.twins.get_my_twin_id(),
+    randomize: true,
+  } as FarmFilterOptions);
+  console.timeEnd("Farms Time");
+
+  if (farms.length < 1) {
+    throw new Error("No farms found");
+  }
+
+  console.time("Total Deployment Time");
+
+  for (let batch = 0; batch < batches; batch++) {
+    console.time("Batch " + (batch + 1));
 
-  for (let i = 0; i < 100; i++) {
-    //Generating the resources
-    const cru = 1;
-    const mru = 256;
-    const diskSize = 5;
-    const rootFs = 1;
-    const vmName = "vm" + generateString(8);
-    const deploymentName = "dep" + generateString(8);
-    const publicIp = false;
-
-    // create network Object
-    const n = new NetworkModel();
-    n.name = "nw" + generateString(5);
-    n.ip_range = "10.238.0.0/16";
-
-    // create disk Object
-    const disk1 = new DiskModel();
-    disk1.name = "d" + generateString(5);
-    disk1.size = diskSize;
-    disk1.mountpoint = "/newDisk1";
-
-    //Farm Selection
-    const farms = await grid3.capacity.filterFarms({
-      nodeMRU: mru / 1024,
-      nodeSRU: diskSize + rootFs,
-      publicIp: publicIp,
-      availableFor: await grid3.twins.get_my_twin_id(),
-      randomize: true,
-    } as FarmFilterOptions);
-
-    if (farms.length < 1) {
-      throw new Error("No farms found");
+    const nodesPromises = Array.from({ length: batchSize }, async () => {
```

Comment:
here we make many requests to the grid proxy to get the nodes. However, we can set all the farm ids we got in the request and just get a random page by using the `randomize` query